web-registry/src/pages/CreditClassDetails/CreditClassDetails.Projects.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
+import Box from '@mui/material/Box';
+
+import { BuySellOrderFlow } from 'features/marketplace/BuySellOrderFlow/BuySellOrderFlow';
+import { ProjectWithOrderData } from 'pages/Projects/Projects.types';
+import { ProjectCardsSection } from 'components/organisms/ProjectCardsSection/ProjectCardsSection';
+import { useProjectsWithOrders } from 'hooks/projects/useProjectsWithOrders';
+
+interface Props {
+  classId: string;
+}
+
+function ProjectsSection({ classId }: Props): JSX.Element {
+  const { projectsWithOrderData } = useProjectsWithOrders({
```

Comment:
This hook uses the `projects` method which query all the projects. In this case, we should use [ProjectsByClass](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.Query.ProjectsByClass) instead.
We can either modify the existing hook here or duplicate it if it's easier.

Reply:
@flagrede Good point, thanks! I just do initial check to launch the specific initial query. I remove post filtering.